topdown/query.go

```diff
@@ -471,6 +471,14 @@ func (q *Query) Run(ctx context.Context) (QueryResultSet, error) {
 // Iter executes the query and invokes the iter function with query results
 // produced by evaluating the query.
 func (q *Query) Iter(ctx context.Context, iter func(QueryResult) error) error {
+	// Query evaluation must not be allowed if the compiler has errors and is in an undefined, possibly inconsistent state
+	if q.compiler != nil && len(q.compiler.Errors) > 0 {
+		return &Error{
+			Code:    InternalErr,
+			Message: "compiler has errors",
```

Comment:
I'm actually a bit uncertain as to what's the best option here. This error will be returned on the `/v1/data` endpoint, and it might be a security risk to forward compilation errors to the querying client .. (or I'm just way to cautious 😅).

The compiler errors are reported for each bundle on the `/v1/status` endpoint.

.. but now that I'm thinking about it, that doesn't help you much when you're not running OPA in server mode 🤔.

Reply:
We can always expose more debug info in the future if we think it's necessary and/or would assist with issue triage.

Reply:
And since these errors will be exposed via the Status API we still have some insight here.